pkg/apis/eventing/v1beta1/trigger_types.go

```diff
@@ -0,0 +1,121 @@
+ * limitations under the License.
+ */
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+const (
+	// DependencyAnnotation is the annotation key used to mark the sources that the Trigger depends on.
+	// This will be used when the kn client creates an importer and trigger pair for the user such that the trigger only receives events produced by the paired importer.
+	DependencyAnnotation = "knative.dev/dependency"
+	// InjectionAnnotation is the annotation key used to enable knative eventing injection for a namespace and automatically create a default broker.
+	// This will be used when the client creates a trigger paired with default broker and the default broker doesn't exist in the namespace
+	InjectionAnnotation = "knative-eventing-injection"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Trigger represents a request to have events delivered to a consumer from a
+// Broker's event pool.
+type Trigger struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the Trigger.
+	Spec TriggerSpec `json:"spec,omitempty"`
+
+	// Status represents the current state of the Trigger. This data may be out of
+	// date.
+	// +optional
+	Status TriggerStatus `json:"status,omitempty"`
+}
+
+var (
+	// Check that Trigger can be validated, can be defaulted, and has immutable fields.
+	_ apis.Validatable = (*Trigger)(nil)
+	_ apis.Defaultable = (*Trigger)(nil)
+
+	// Check that Trigger can return its spec untyped.
+	_ apis.HasSpec = (*Trigger)(nil)
+
+	_ runtime.Object = (*Trigger)(nil)
+
+	// Check that we can create OwnerReferences to a Trigger.
+	_ kmeta.OwnerRefable = (*Trigger)(nil)
+)
+
+type TriggerSpec struct {
+	// Broker is the broker that this trigger receives events from. If not specified, will default
+	// to 'default'.
+	Broker string `json:"broker,omitempty"`
+
+	// Filter is the filter to apply against all events from the Broker. Only events that pass this
+	// filter will be sent to the Subscriber. If not specified, will default to allowing all events.
+	//
+	// +optional
+	Filter *TriggerFilter `json:"filter,omitempty"`
+
+	// Subscriber is the addressable that receives events from the Broker that pass the Filter. It
+	// is required.
+	Subscriber duckv1.Destination `json:"subscriber"`
+}
+
+type TriggerFilter struct {
+	// Attributes filters events by exact match on event context attributes.
+	// Each key in the map is compared with the equivalent key in the event
+	// context. An event passes the filter if all values are equal to the
+	// specified values.
+	//
+	// Nested context attributes are not supported as keys. Only string values are supported.
+	//
+	// +optional
+	Attributes *TriggerFilterAttributes `json:"attributes,omitempty"`
```

Comment:
does this need to be a pointer if `TriggerFilterAttributes` is already a `map[]` ?

Reply:
Nope, you're right. for map no need to have a pointer ✅